src/tcp_address.cpp

```diff
@@ -622,33 +647,6 @@ int zmq::tcp_address_t::resolve (const char *name_, bool local_, bool ipv6_, boo
           addr_str [addr_str.size () - 1] == ']')
         addr_str = addr_str.substr (1, addr_str.size () - 2);
 
-    // Test the '%' to know if we have an interface name / zone_id in the address
```

Comment:
To avoid risks of regressions on weird platforms, I think it would be better to leave this block in place, but use "zone_id" below only if the [source_]address.ipv6.sin6_scope_id variables are 0, so that the value set by the system API has the precedence